packages/react/hover-card/src/HoverCard.tsx

```diff
@@ -256,6 +263,50 @@ const HoverCardContentImpl = React.forwardRef<
   const popperScope = usePopperScope(__scopeHoverCard);
   const ref = React.useRef<HoverCardContentImplElement>(null);
   const composedRefs = useComposedRefs(forwardedRef, ref);
+  const [containTextSelection, setContainTextSelection] = React.useState(false);
+
+  React.useEffect(() => {
+    if (containTextSelection) {
+      const body = document.body;
+
+      // Safari requires prefix
+      originalBodyUserSelect = body.style.userSelect || body.style.webkitUserSelect;
+
+      body.style.userSelect = 'none';
+      body.style.webkitUserSelect = 'none';
+      return () => {
+        body.style.userSelect = originalBodyUserSelect;
+        body.style.webkitUserSelect = originalBodyUserSelect;
+      };
+    }
+  }, [containTextSelection]);
+
+  const { onDismiss } = context;
+
+  React.useEffect(() => {
+    if (ref.current) {
+      const handlePointerUp = (event: PointerEvent) => {
+        // Delay a frame to ensure we always access the latest selection
+        setTimeout(() => {
+          const hasEmptySelection = document.getSelection()?.toString() === '';
+          setContainTextSelection(false);
+
+          if (hasEmptySelection) {
+            const isPointerUpOutsideContent = !ref.current?.contains(event.target as HTMLElement);
+            context.isTextSelectingRef.current = false;
+            if (isPointerUpOutsideContent) onDismiss();
```

Comment:
It seems native closes on mousemove outside?

Reply:
Are you referring to the VSC example? or another?

Reply:
Yes VSC

Reply:
Gotcha, I had a version that did this and found in practical terms it added little as the card dismisses immediately if you're at all moving the pointer following `pointerUp` (which I assume would be most of the time unless you intentionally try not to)

Reply:
Yeah I did wonder, because it unmounts, you wouldn't be able to copy the text if you haven't done so before moving your mouse. So probably better to leave it as is?

Reply:
> So probably better to leave it as is?

Yeah I think what we have is good enough